### PR TITLE
Indirect FQ simultaneous fitting

### LINKED ISSUE #26881: Convolution fit and simultaneous fitting
simultaneous fitting in the indirect data analysis convolution fit tab

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
**Description of work.**

This adds the simultaneous fitting to the FQ tab in mantid. 

**To test:**
  * If testing this PR slack me and I'll send you the relevant data file.
  * Load data file into FQ tab and try the a couple of relevant fits. Check that fits look correct.
  * Try switching between full function tab and template function tab.
  * Try some weird things to try and break the GUI

Fixes #26881  

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/scientific_interfaces/Indirect/IndirectFunctionBrowser/FQTemplatePresenter.cpp [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+                                                 double value) {
+  m_model.setLocalParameterValue(parName, i, value);
+}
+
+void FQTemplatePresenter::setLocalParameterTie(const QString &parName, int i,
+                                               const QString &tie) {
+  m_model.setLocalParameterTie(parName, i, tie);
+}
+
+void FQTemplatePresenter::updateView() { updateViewParameters(); }
```
Comment (Harrietbrown) [inline qt/scientific_interfaces/Indirect/IndirectFunctionBrowser/FQTemplatePresenter.cpp]: Why does this need to be a separate function

**Review by Harrietbrown — COMMENTED** [review]
There is still that issue with error message coming up if the gui is too small, but that is already a seperate issue

File: qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp [hunk 2]
```diff
@@ -696,9 +697,10 @@ QStringList IndirectFitAnalysisTab::getDatasetNames() const {
     TableDatasetIndex index{i};
     auto const name =
         QString::fromStdString(m_fittingModel->getWorkspace(index)->getName());
-    auto const numberSpectra = m_fittingModel->getNumberOfSpectra(index);
-    for (int j{0}; j < numberSpectra.value; ++j) {
-      datasetNames << name + " (" + QString::number(j) + ")";
+    m_fittingModel->getNumberOfSpectra(index);
```
[the remark this answers is not in the document]
Comment (Mantid-Matthew) [reply qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp]: Good spot I don't think that line is needed at all

File: qt/scientific_interfaces/Indirect/IndirectFunctionBrowser/FQTemplatePresenter.cpp (hunk 1 above)
Comment (Mantid-Matthew) [reply qt/scientific_interfaces/Indirect/IndirectFunctionBrowser/FQTemplatePresenter.cpp]: It doesn't was just hanging around from legacy code. I've now removed updateViewParameters.

**Comment by Harrietbrown** [comment]
There are a lot of indirect tests failing on python 2, should you have a look into them

**Comment by Mantid-Matthew** [comment]
I have had a look and this seems like a flaky indirect test from a part of the code not touched by this PR. I've therefore hit rebuild to see what it does.

**Review by Harrietbrown — APPROVED** [review]
The F(Q) fitting tab on indirect functions as expected, the fits look correct and the plots appear as expected, switching between function displays also functions properly.

Good job

**Pull request merged**